src/pybind/mgr/dashboard/frontend/src/app/shared/api/rbd.service.spec.ts

```diff
@@ -0,0 +1,206 @@
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { fakeAsync, inject, TestBed, tick } from '@angular/core/testing';
+
+import 'rxjs/add/observable/of';
+import { Observable } from 'rxjs/Observable';
+
+import { RbdService } from './rbd.service';
+
+describe('RbdService', () => {
+  let count = 0;
+  const fake = () => {
+    count++;
+    return Observable.of();
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [RbdService],
+      imports: [HttpClientTestingModule]
+    });
+
+    const httpClient = TestBed.get(HttpClient);
+    spyOn(httpClient, 'get').and.callFake(fake);
+    spyOn(httpClient, 'post').and.callFake(fake);
+    spyOn(httpClient, 'put').and.callFake(fake);
+    spyOn(httpClient, 'delete').and.callFake(fake);
+
+    count = 0;
+  });
+
+  it(
+    'should be created',
+    inject([RbdService], (service: RbdService) => {
+      expect(service).toBeTruthy();
+    })
+  );
+
+  it(
+    'should call create',
+    inject(
+      [RbdService],
+      fakeAsync((service: RbdService) => {
+        expect(service.create(null)).toEqual(jasmine.any(Observable));
+        tick();
+        expect(count).toBe(1);
+      })
+    )
+  );
+
+  it(
+    'should call delete',
+    inject(
+      [RbdService],
+      fakeAsync((service: RbdService) => {
+        expect(service.delete(null, null)).toEqual(jasmine.any(Observable));
+        tick();
+        expect(count).toBe(1);
+      })
+    )
+  );
+
+  it(
+    'should call update',
+    inject(
+      [RbdService],
+      fakeAsync((service: RbdService) => {
+        expect(service.update(null, null, null)).toEqual(jasmine.any(Observable));
+        tick();
+        expect(count).toBe(1);
+      })
+    )
+  );
+
+  it(
+    'should call get',
+    inject(
+      [RbdService],
+      fakeAsync((service: RbdService) => {
+        expect(service.get(null, null)).toEqual(jasmine.any(Observable));
+        tick();
+        expect(count).toBe(1);
+      })
+    )
+  );
+
+  it(
+    'should call list',
+    inject(
+      [RbdService],
+      fakeAsync((service: RbdService) => {
+        expect(service.list()).toEqual(jasmine.any(Observable));
+        tick();
+        expect(count).toBe(1);
+      })
+    )
+  );
+
+  it(
+    'should call copy',
+    inject(
+      [RbdService],
+      fakeAsync((service: RbdService) => {
+        expect(service.copy(null, null, null)).toEqual(jasmine.any(Observable));
```

Comment:
It would make sense not just to test if `http` works as expected but to test which URL it will use based on the input parameters. The behavior should be tested.

(This comment will also apply to many other tests)

Reply:
Done